### PR TITLE
[For review] New ctfe

### PR DESCRIPTION
**Pull request by UplinkCoder**
Just for review.
I cannot seem to revive the old PR. :(

### CODE REVIEW

**Comment by MartinNowak** [comment]
The old PR being #5894.

> I cannot seem to revive the old PR. :(

What does that mean? Anyhow, looks fine to squash and just make a new one.

File: src/bc.d [hunk 1]
```diff
@@ -0,0 +1,1817 @@
+module ddmd.ctfe.bc;
+import ddmd.ctfe.bc_common;
+import ddmd.ctfe.bc_builtins;
+import core.stdc.stdio;
+import std.conv;
+
+/**
+ * Written By Stefan Koch in 2016
+ * All Rights Reserved.
```
Comment (mathias-lang-sociomantic) [inline src/bc.d]: Check the README, copyright should be assigned.

**Review by mathias-lang-sociomantic — COMMENTED** [review]
Would be nice to havea high level view of the design of this new engine.
I guess you are planning on adding detailed documentation later ?

**Comment by UplinkCoder** [comment]
I just updated to the latest WIP.
I don't know why it won't build...

File: src/posix.mak [hunk 2]
```diff
@@ -237,11 +237,12 @@ else
     FRONT_SRCS += libelf.d scanelf.d
 endif
 
-GLUE_SRCS=$(addsuffix .d, irstate toelfdebug toctype gluelayer todt tocsym toir)
+GLUE_SRCS=$(addsuffix .d, irstate toelfdebug toctype glue gluelayer todt tocsym toir dmsc \
```
Comment (yebblies) [inline src/posix.mak]: This is why it doesn't build.  There is no glue.d in this branch.  Most of the changes in this file should be removed.

Comment (UplinkCoder) [inline src/posix.mak]: Ah Thanks

**Comment by ZombineDev** [comment]
@UplinkCoder can you rebase your changes on top of master and push them to this branch? I would like to see your engine tested on the various CIs, plus it makes easier to review individual commits when you know that each one builds cleanly.

**Comment by ZombineDev** [comment]
BTW, after you get your branch to building cleanly, I would strongly recommend to use `git commit --amend` for fixup changes on new commits (such as whitespace, typos and style). This is the recommended practice on the various D projects - have each commit build cleanly, conform to the coding style and be small enough to be easy to understand and review in isolation.

**Comment by UplinkCoder** [comment]
@MartinNowak I consider this ready for merge into the feature branch.
As long as it cannot pass the unittests it has to be enabled using the -bc-ctfe switch

**Comment by MartinNowak** [comment]
> BTW, after you get your branch to building cleanly, I would strongly recommend using git commit --amend for fixup changes on new commits

We can cleanup git history later, and it won't hurt us much when the mess is behind a single merge request.
As a practical problem many people have an aversion to git, simply never spent the time to train workflows/automate, or just try to get along with one of the (over-)simplified GUIs (including github).
As for getting this important feature done, we decided that proper git usage should not be a criteria for feature branch reviews. Those should rather be streamlined for development workflow.
Fixing whitespace, indentation, git history, nitbits can all be done when merging back the feature branch into master.

**Comment by MartinNowak** [comment]
Coarse review is planned for next week, once 2.071.1 is out.

**Comment by MartinNowak** [comment]
Just merged the first chunk up to bc05f67567b9689324fd5e6f32e107f13c965afb, that's about as much as I can still overview. Let's have another review/call for the things added after that. I'm seeing some exiting news about struct support and function calls :).

**Comment by UplinkCoder** [comment]
@MartinNowak Struct Support is done so far.
FunctionCalls however are going to keep me busy for a while.

**Comment by MartinNowak** [comment]
Just merged everything up to 5943bac5e641c6258902a12e7000b4497b5a7e21.

**Pull request merged**